src/cmd-cloud-prune

```diff
@@ -0,0 +1,303 @@
+                                delete_cloud_resources(currentBuild, gcp_cloud_config, args.dry_run)
+                                if not args.dry_run:
+                                    build.setdefault("policy-cleanup", []).append(action)
+                                    buildJsonData["builds"][index] = build
+                            # Not implemented yet
+                            case "build":
+                                print(f"Deleting key {prefix}{build.id} from bucket {bucket}")
+                                # Delete the build's directory in S3
+                                # S3().delete_object(args.bucket, f"{args.prefix}{str(currentBuild.id)}")
+                else:
+                    break
+        if not args.dry_run:
+            with open("builds/builds.json", "w") as json_file:
+                json_file.write(json.dumps(buildJsonData))
+            # This copies the local builds.json and updates the S3 bucket version.
+            s3_copy(s3_client, BUILDFILES['list'], bucket, f'{prefix}/builds/builds.json',
+                    CACHE_MAX_AGE_METADATA, args.acl, extra_args={},
+                    dry_run=args.dry_run)
+
+
+def get_json_from_s3(bucket, key):
+    # Create an S3 client
+    s3 = boto3.client("s3")
+
+    try:
+        # Fetch the JSON file from S3
+        response = s3.get_object(Bucket=bucket, Key=key)
+
+        # Read the content of the file
+        content = response["Body"].read().decode("utf-8")
+
+        # Parse the JSON content
+        json_content = json.loads(content)
+
+        return json_content
+
+    except Exception as e:
+        print(f"Error fetching the JSON file from S3 {bucket}/{key}: {e}")
+        return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="coreos-assembler cloud-prune")
+    parser.add_argument("--policy", default='./policy.yaml', type=str,
+                        help="Path to policy.yaml file")
+    parser.add_argument("--dry-run", help="Don't actually delete anything",
+                        action='store_true')
+    parser.add_argument("--stream", type=str, help="Fedora stream", required=True)
+
+    parser.add_argument("--gcp-json-key", help="GCP Service Account JSON Auth",
+                        default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--gcp-project", help="GCP Project name",
+                        default=os.environ.get("GCP_PROJECT_NAME"))
+
+    subparsers = parser.add_subparsers(dest="cmd", title="subcommands")
+    subparsers.required = True
+
+    s3 = subparsers.add_parser('s3', help='Prune s3 buckets')
+    s3.add_argument("url", metavar='<BUCKET>[/PREFIX]',
+                    help="Bucket and path prefix in which to upload")
+    s3.add_argument("--acl", help="ACL for objects",
+                    action='store', default='private')
+    s3.add_argument("--aws-config-file", default=os.environ.get("AWS_CONFIG_FILE"),
+                    help="Path to AWS config file")
```

Comment:
Why is this a subparser?

Reply:
I took this from how we get s3 arguments in [buildupload](https://github.com/coreos/coreos-assembler/blob/main/src/cmd-buildupload#L57)